cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerService.java

```diff
@@ -76,6 +167,11 @@ protected void shutDown() throws Exception {
     LOG.debug("Shutting down ArtifactLocalizerService has completed");
   }
 
+  @Override
+  protected Scheduler scheduler() {
+    return Scheduler.newFixedDelaySchedule(0, cacheCleanupInterval, TimeUnit.MINUTES);
+  }
```

Comment:
Shouldn't we also make sure that we don't evict artifact cache that is being used by a taskworker?

Reply:
Changed the initial delay to be `cacheCleanupInterval`.

The task worker should only be using the newest artifact at any given time. Since we only allow one request at time there is no possibility of the artifact getting updated while the task is running (since we only update artifacts when the localizer receives a request) so it should always be safe to delete an out of date artifact.